### PR TITLE
[quant][graphmode] Returning axis from calculate_qparams

### PR DESCRIPTION
**Pull request by jerryzh168**
Stack from [ghstack](https://github.com/ezyang/ghstack):
* **#29494 [quant][graphmode] Returning axis from calculate_qparams**
* #29493 [quant][graphmode] Test for per channel graph mode quantization
* #29416 [quant] Make PerChannelMinMaxObserver scriptable using `torch.jit.ignore`

Summary:
`calculate_qparams` of per channel quantization should return the axis, this
PR added this and also added corresponding support in graph mode

Test Plan:
python test/test_jit.py

Reviewers:
mvz

Subscribers:

Tasks:

Tags:

Differential Revision: [D18580905](https://our.internmc.facebook.com/intern/diff/D18580905)

### CODE REVIEW

File: torch/quantization/observer.py [hunk 1]
```diff
@@ -134,7 +134,7 @@ def _calculate_per_channel_qparams(self, min_vals, max_vals):
                 "must run observer before calling calculate_qparams.\
                                     Returning default scale and zero point "
             )
-            return torch.tensor([1.0]), torch.tensor([0])
+            return torch.tensor([1.0]), torch.tensor([0]), self.ch_axis
```
Comment (raghuramank100) [inline torch/quantization/observer.py]: Not sure I follow the reasoning here: The axis is passed as an attribute to observer and is used when statistics are collected. CalcQparams should be independent of the axis argument, it should know if per channel or per tensor calculation is desired, but operations are independent of the axis argument.

Comment (jerryzh168) [reply torch/quantization/observer.py]: we don't need axis for calculate_qparams, but we need the axis argument for per channel quantize op. so it's an output value, not an input argument to calculate_qparams.

Comment (jerryzh168) [reply torch/quantization/observer.py]: we'll have a separate `get_qparam` function that returns these things, I'll do this in a separate PR

**Review by raghuramank100 — APPROVED** [review]

File: torch/csrc/jit/passes/quantization.cpp [hunk 2]
```diff
@@ -403,8 +403,8 @@ Node* insertQuantDeQuantCall(
   if (is_per_channel) {
     scale_val = g->insertConstant(scale);
     zero_point_val = g->insertConstant(zero_point);
-    // TODO: get axis from qparam
-    axis_val = g->insertConstant(0);
+    int axis = tp->elements()[2].toInt();
```
Comment (ZolotukhinM) [inline torch/csrc/jit/passes/quantization.cpp]: This should be squashed with the change where it was introduced before landing. There is no reason to commit half-finished stuff.

Comment (jerryzh168) [reply torch/csrc/jit/passes/quantization.cpp]: this is just because the order we do things right now, at the place we introduce this, calculate_qparam doesn't return axis

Comment (ZolotukhinM) [reply torch/csrc/jit/passes/quantization.cpp]: Ok, I reread everything more carefully and it makes sense.

**Review by ZolotukhinM — APPROVED** [review]
